### ISSUE TITLE
Any edit to a line of code causes an undefined Name error

### ISSUE THREAD

**Issue by bhc3** [jupyter/notebook#361]
I'm going through the Apache Spark Starter Tutorial 1: Load Data.ipynd via the Bluemix portal. I decided to alter a line of code in the tutorial, to test the functionality. For the 'weather' element, there is a code line that prints the first record in the file: weather.first().

I decided to change that function to one that worked on the preceding code line: weather.count(). I made the change and clicked the Run arrow button to update the Notebook.

Rather than print the record count, I got an error telling me the 'weather' name was not defined. Here's the actual text:

---------------

NameError Traceback (most recent call last)
in ()
----> 1 print "The first line in the 2015.csv dataset:", weather.count()

NameError: name 'weather' is not defined

---------------

I'm including a screenshot of the error. Not sure why the 'weather' name is recognized on initial load, but becomes undefined on a simple edit.

![jupyter notebook error - undefined name](https://cloud.githubusercontent.com/assets/4078075/9553050/6172d3e4-4d71-11e5-815c-5915a04a4b5d.png)

**Comment by Carreau** [comment]
Can you give the link where you got this tutorial. This is just the repository for the notebook App, so if there is an issue it's most likely a problem with the tutorial, not with the notebook itself. 

It might be a problem in the notebook application itself, I would be happy to test. But without a link to the tutorial or the notebook itself, it next to impossible for me. 

Google does not seem to find out where this is. Do you have a hint of the author tutorial, we might be able to contact the person at IBM to sort that out.

**Comment by bhc3** [comment]
Hi Carreau -

Thanks for your response. The notebook tutorial is accessible after you set up an app within the Bluemix environment. You need to have a unique app URL to get to it. App access is password protected, so I can't give you access through here.

I could run a video showing what I'm doing, if that would help. Just let me know. Otherwise, this may be an issue the IBM team needs to investigate.

Hutch

**Comment by Carreau** [comment]
Hum let me try to ping some IBM people first:

@vinomaster, @lbustelo and @parente are IIRC working on IBM and with spark, one of them might be better suited to respond.  If they can't help you I'll try to take some time to create an account and see what i can do.

**Comment by bhc3** [comment]
Hi Carreau -

Update this morning. I went to the notebook and tried some extra things before I made my first change. I clicked the Run All right at the outset. I then went into a couple different cells and clicked Run specifically for them. I was testing to see if the commands would break without any editing. They did not.

I then made the change I've described in this issue. I changed the first() function to a count(function). I then clicked Run for that cell. This time, it worked. No NameError.

The biggest difference I've done this morning is that I specifically clicked the menu Cell > Run All, rather than the "play" arrow button which only runs the specific cell. My sense is that the notebook's set of commands are not Run on page load, but await my first time clicking Run All. It was that lack of the initial Run All that caused the NameError of the 'weather' file.

If so, I've learned a bit about the workings of the Jupyter notebook. Thanks so much for your time and attention.

Hutch
